protocol-units/bridge/shared/tests/bridge_service.rs

```diff
@@ -0,0 +1,267 @@
+	let transfer_initiated_event = bridge_service.next().await.expect("No event");
+	let transfer_initiated_event =
+		transfer_initiated_event.B1I_ContractEvent().expect("Not a B1I event");
+	tracing::debug!(?transfer_initiated_event);
+	assert_eq!(
+		transfer_initiated_event,
+		&BridgeContractInitiatorEvent::Initiated(BridgeTransferDetails {
+			bridge_transfer_id: transfer_initiated_event.bridge_transfer_id().clone(),
+			initiator_address: InitiatorAddress(BC1Address("initiator")),
+			recipient_address: RecipientAddress::from(BC1Address("recipient")),
+			hash_lock: HashLock(BC1Hash::from("hash_lock")),
+			time_lock: TimeLock(100),
+			amount: Amount(1000)
+		})
+	);
+
+	// Step 2: Locking the assets on the Blockchain 2
+
+	// Upon recognizing the event, our bridge server has invoked the counterparty
+	// contract on blockchain 2 to initiate asset locking within the smart contract.
+	tracing::debug!("Locking assets on Blockchain 2");
+
+	let counterparty_locked_event = bridge_service.next().await.expect("No event");
+	let counterparty_locked_event =
+		counterparty_locked_event.B2C_ContractEvent().expect("Not a B2C event");
+	tracing::debug!(?counterparty_locked_event);
+	assert_eq!(
+		counterparty_locked_event,
+		&BridgeContractCounterpartyEvent::Locked(LockDetails {
+			bridge_transfer_id: Convert::convert(transfer_initiated_event.bridge_transfer_id()),
+			hash_lock: HashLock(BC2Hash::from("hash_lock")),
+			time_lock: TimeLock(100),
+			recipient_address: RecipientAddress::from(BC2Address("recipient")),
+			amount: Amount(1000),
+		})
+	);
+
+	// Step 3: Client completes the swap on Blockchain 2, revealing the pre_image of the hash lock
+
+	// Once the assets are secured within the counterparty smart contract, the initiator is able
+	// to execute the complete bridge transfer by disclosing the secret key required to unlock the assets.
+	tracing::debug!("Client completing bridge transfer on Blockchain 2");
+
+	<B2Client as BridgeContractCounterparty>::complete_bridge_transfer(
+		&mut blockchain_2_client,
+		Convert::convert(transfer_initiated_event.bridge_transfer_id()),
+		HashLockPreImage(b"hash_lock".to_vec()),
+	)
+	.await
+	.expect("complete_bridge_transfer failed");
+
+	// As the claim was made by the counterparty, we anticipate the bridge to generate a bridge
+	// contract counterpart event.
+	let completed_event_counterparty = bridge_service.next().await.expect("No event");
+	let completed_event_counterparty =
+		completed_event_counterparty.B2C_ContractEvent().expect("Not a B2C event");
+	tracing::debug!(?completed_event_counterparty);
+	assert_eq!(
+		completed_event_counterparty,
+		&BridgeContractCounterpartyEvent::Completed(CompletedDetails {
+			bridge_transfer_id: Convert::convert(transfer_initiated_event.bridge_transfer_id()),
+			recipient_address: RecipientAddress::from(BC2Address("recipient")),
+			hash_lock: HashLock(BC2Hash::from("hash_lock")),
+			secret: HashLockPreImage(b"hash_lock".to_vec()),
```

Comment:
"Preimage" is one word in mathematics and cryptography. I don't think the camel-case is needed here.

Reply:
I have included the case for clarity. Would you prefer it to be renamed to HashLockPreimage? The same logic could also be applied to a HashLock for example. You would also consider that one word. Please give me your preferences and I will rename it towards whatever you like.

Reply:
It's a nit.